### PR TITLE
[XPU] support ignore_index in xpu c_softmax_with_cross_entropy 

### PR DESCRIPTION
**Pull request by runzhech**
### PR Category
Custom Device

### PR Types
Others

### Description
support ignore_index in xpu's c_softmax_with_cross_entropy. Drop the logits of label which should be ignored and make corresponding change in backward propagation.

### CODE REVIEW

**Review by houj04 — COMMENTED** [review]
前向部分的修改，似乎没有用到`ignore_index`这个输入？-100只是一个默认值，当这个值喂进来一个正数的时候，也要应该能正确ignore掉才行。

**Review by XiaociZhang — COMMENTED** [review]
之前关闭ignore_index的地方没改
L43
L560
